charts/server/templates/pod-monitor.yaml

```diff
@@ -1,16 +1,19 @@
-{{- if .Values.metrics.enabled }}
+{{- if and .Values.metrics.enabled .Values.prometheus.podmonitor.enabled ( .Capabilities.APIVersions.Has "monitoring.coreos.com/v1" ) }}
 apiVersion: monitoring.coreos.com/v1
 kind: PodMonitor
 metadata:
   name: {{ include "woodpecker-server.fullname" . }}
   labels:
     {{- include "woodpecker-server.labels" . | nindent 4 }}
+    {{- with .Values.prometheus.podmonitor.labels }}
+    {{- toYaml . | nindent 4}}
+    {{- end }}
 spec:
   selector:
     matchLabels:
-    {{- include "woodpecker-server.selectorLabels" . | nindent 6 }}
+      {{- include "woodpecker-server.selectorLabels" . | nindent 6 }}
   podMetricsEndpoints:
-  - port: metrics
-    interval: {{ .Values.metrics.interval }}
```

Comment:
Should we keep the `interval` parameter? What is the default when unsetting it?

Reply:
i just make th indent more beautiful - we keep interval and put that under `prometheus.podmonitor.interval` in `values.yaml`

Reply:
you are correct, i make it optional

the default value of an prometheus server is 1m, this global scrape_interval could be changed, and that value could be overwritten by this Monitor :)